#!/usr/bin/env python
PACKAGE = "data_base_recorder"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("save_location",    str_t,    0, "Physical address of folder to save the dataset",  "/mnt/ameli/live_dataset")
gen.add("start_recording",   bool_t,   0, "If not true data is not saved even if aruco marker not detected",  True)
size_enum = gen.enum([ gen.const("perspective_transform", str_t, "/perspective_transformation/image", "perspective transform"),
                       gen.const("raw_image", str_t, "/usb_cam/image_raw", "usb_cam raw image"),
                       gen.const("compressed_image", str_t, "/usb_cam/compressed", "usb_cam compressed image")],
                       "List of subscribable topics")

gen.add("topic_to_record", str_t, 0, "Topic to be recorded", "/perspective_transformation/image", edit_method=size_enum)

exit(gen.generate(PACKAGE, "data_base_recorder", "Recorder"))